pipeline/data_store.py

```diff
@@ -1382,8 +1382,8 @@ def delete_everything(self, session=None):
             autoflush_state = session.autoflush
             try:
                 obj_list = self.get_all_data_products(output='list', omit_exposure=True)
-                for obj in obj_list:  # first make sure all are merged
-                    obj = obj.recursive_merge(session)
+                for i, obj in enumerate(obj_list):  # first make sure all are merged
+                    obj_list[i] = obj.recursive_merge(session)
```

Comment:
Aha!  This is one of those fixes I see and think "duh, of course"... but python sometimes lulls us into a false sense of security with all its high-level syntactic sugar.  Teaching computational physics to students who aren't experienced programmers, and haven't done anything in a lower-level language like C before, it can be a little bit tricky to explain what's going on here.